visualization/app/codeCharta/util/fileDownloader.ts

```diff
@@ -1,71 +1,62 @@
-            blacklist: file.settings.fileSettings.blacklist,
-        };
-    }
-
-    private static removeJsonHashkeysAndVisibleAttribute(map: CodeMapNode) {
-        let copy = JSON.parse(JSON.stringify(map));
-        d3.hierarchy(copy).each((node)=>{
-            delete node.data.visible;
-        });
-        return copy;
-    }
-
-    private static getNewFileName(file: CCFile) {
-        const datedFileName = this.addDateToFileName(file.fileMeta.fileName);
-        return this.addJsonFileEndingIfNecessary(datedFileName);
-    }
-
-    private static addDateToFileName(fileName) {
-        const date = new Date();
-        const dateString = date.getDate() + "_" + (date.getMonth() + 1)  + "_" + date.getFullYear();
-        let tokens = fileName.split(".");
-        tokens.splice(1, 0, dateString);
-        return tokens.join(".");
-    }
-
-    private static addJsonFileEndingIfNecessary(fileName) {
-        if(!fileName.endsWith(".json")) {
-            return fileName + ".json";
-        }
-        return fileName;
-    }
-
-    private static downloadData(data, fileName) {
-        let dataJson = data;
-        if (typeof data === "object") {
-            dataJson = angular.toJson(data, 4);
-        }
-
-        const blob = new Blob([dataJson], {type: "text/json"});
-        const e = document.createEvent("MouseEvents");
-        const a = document.createElement("a");
-
-        a.download = fileName;
-        a.href = window.URL.createObjectURL(blob);
-        a.dataset.downloadurl = ["text/json", a.download, a.href].join(":");
-        e.initMouseEvent("click", true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
-        a.dispatchEvent(e);
-    }
-
+	public static downloadCurrentMap(file: CCFile) {
+		const data = this.getProjectDataAsCCJsonFormat(file)
+		this.downloadData(data, this.getNewFileName(file))
+	}
+
+	private static getProjectDataAsCCJsonFormat(file: CCFile) {
+		let newFileName = this.getNewFileName(file)
+
+		return {
+			fileName: newFileName,
+			projectName: file.fileMeta.projectName,
+			apiVersion: file.fileMeta.apiVersion,
+			nodes: [this.removeJsonHashkeysAndVisibleAttribute(file.map)],
+			edges: file.settings.fileSettings.edges,
+			attributeTypes: file.settings.fileSettings.attributeTypes,
+			blacklist: file.settings.fileSettings.blacklist
+		}
+	}
+
+	private static removeJsonHashkeysAndVisibleAttribute(map: CodeMapNode) {
+		let copy = JSON.parse(JSON.stringify(map))
+		d3.hierarchy(copy).each(node => {
+			delete node.data.visible
+		})
+		return copy
+	}
+
+	private static getNewFileName(file: CCFile) {
+		return this.addDateToFileName(file.fileMeta.fileName)
+	}
+
+	private static addDateToFileName(fileName) {
+		const date = new Date()
+		const dateString =
+			date.getFullYear() + "-" + (date.getMonth() + 1) + "-" + date.getDate() + "_" + date.getHours() + "-" + date.getMinutes()
+		let firstToken = fileName.split(".")[0]
```

Comment:
If the filename was `myPrefix.codecharta.cc.json`
the new filename were `myPrefix.YYYY-mm-dd_HH-MM.cc.json` 
you need to check a bit more than just the firstToken, as there might be more points in a filename. I would suggest using a regular expression, which replaces the date, if the fileName has one. Otherwise you need to subtract the `.json` or `.cc.json`.

Reply:
And you could add some more tests like e.x.:
- `prefix.name.suffix.cc.json` -> `prefix.name.suffix.2019-04-30_15-00.cc.json`
- `prefix.name.suffix.json` -> `prefix.name.suffix.2019-04-30_15-00.cc.json`
- `prefix.name.suffix.2000-01-01_01-01.cc.json` -> `prefix.name.suffix.2019-04-30_15-00.cc.json`
- `prefix.name.suffix.2000-01-01_01-01.json` -> `prefix.name.suffix.2019-04-30_15-00.cc.json`